app/main.py

```diff
@@ -1 +1,27 @@
-# write your code here
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = False
+        self.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, obj: Animal) -> None:
+        if isinstance(obj, Herbivore) and not obj.hidden:
+            obj.health -= 50
+            if obj.health <= 0:
+                Animal.alive.remove(obj)
```

Comment:
Let's give more descriptive name for variable. Use `animal`, or `victim` for example